concert/tests/unit/test_parameter.py

```diff
@@ -194,6 +212,17 @@ def test_saving(self):
         with self.assertRaises(ParameterError):
             device['no_write'].restore().join()
 
+    def test_saving_with_target_value(self):
+        device = FooDeviceTagetValue(0 * q.mm)
+        device['foo'].upper = None
+        device['foo'].lower = None
+        device['foo'].stash().join()
+        self.assertEqual(device['foo'].target_readable, True)
+        device.foo = 1 * q.mm
+        self.assertEqual(device._value, 1 * q.mm)
```

Comment:
Please always test the user attributes, not the hidden ones. This will immediately force you to get rid of the random number above.
Also, please test the combination of user + external limits and the case that it's not implemented (the proper exception should be raised).